ratelimits/names.go

```diff
@@ -35,26 +46,93 @@ const (
 	// is the registration id of the account.
 	FailedAuthorizationsPerAccount
 
-	// CertificatesPerNamePerAccount uses bucket key 'enum:regId:name', where
-	// name is the a name in a certificate issued to the account matching regId.
-	CertificatesPerNamePerAccount
+	// CertificatesPerDomainPerAccount uses bucket key 'enum:regId:domain',
+	// where name is the a name in a certificate issued to the account matching
+	// regId.
+	CertificatesPerDomainPerAccount
 
-	// CertificatesPerNameSetPerAccount uses bucket key 'enum:regId:nameSet',
+	// CertificatesPerFQDNSetPerAccount uses bucket key 'enum:regId:fqdnSet',
 	// where nameSet is a set of names in a certificate issued to the account
 	// matching regId.
-	CertificatesPerNameSetPerAccount
+	CertificatesPerFQDNSetPerAccount
 )
 
 // nameToString is a map of Name values to string names.
 var nameToString = map[Name]string{
 	UsageRequestsPerIPv4Address:      "UsageRequestsPerIPv4Address",
+	UsageRequestsPerIPv6Range:        "UsageRequestsPerIPv6Range",
 	InfoRequestsPerIPv4Address:       "InfoRequestsPerIPv4Address",
+	InfoRequestsPerIPv6Range:         "InfoRequestsPerIPv6Range",
 	NewRegistrationsPerIPv4Address:   "NewRegistrationsPerIPv4Address",
 	NewRegistrationsPerIPv6Range:     "NewRegistrationsPerIPv6Range",
```

Comment:
Why are we keeping these as two separate limits? Why not a single limit that takes either an IPv4 address or an IPv6 range as its bucket (and override) key?

Reply:
> Why are we keeping these as two separate limits? Why not a single limit that takes either an IPv4 address or an IPv6 range as its bucket (and override) key?

For the purpose of override validation I'd prefer to have a clear expectation of the `Id` value (the former is a single IPv4 address while the latter is expected to be an IPv6 range expressed in CIDR notation). Is your concern that this will complicate our use-case once it is actually implemented?

Reply:
The current implementation crawls registrations looking for IPv6 addresses that fall within the given date and IP range. The key-value implementation will instead increment the TAT for two buckets:
1. The bucket for the specific address.
2. The bucket for the /48 IPv6 range in CIDR notation.

```go
package main

import (
	"fmt"
	"net"
)

func main() {
	input := "2001:db8:1234:5678:90ab:cdef:1234:5678"
	ip := net.ParseIP(input)

	// Truncate IPv6 address to /48 range.
	mask := net.CIDRMask(48, 128)
	rangeForAddr := &net.IPNet{IP: ip.Mask(mask), Mask: mask}

	fmt.Printf("Does IPNet %q contain %q?: %v", rangeForAddr.IP, ip, rangeForAddr.Contains(ip))
}
```